api/types/tunnel.go

```diff
@@ -179,40 +180,23 @@ func (r *ReverseTunnelV2) Check() error {
 	if strings.TrimSpace(r.Spec.ClusterName) == "" {
 		return trace.BadParameter("Reverse tunnel validation error: empty cluster name")
 	}
-
 	if len(r.Spec.DialAddrs) == 0 {
 		return trace.BadParameter("Invalid dial address for reverse tunnel '%v'", r.Spec.ClusterName)
 	}
-
-	for _, addr := range r.Spec.DialAddrs {
-		if _, err := utils.ParseAddr(addr); err != nil {
+	for _, addr := range r.GetDialAddrs() {
+		if addr == "" {
+			return trace.BadParameter("missing parameter address")
+		}
+		if !strings.Contains(addr, "://") {
+			return nil
+		}
+		if _, err := url.Parse(addr); err != nil {
```

Comment:
Something that's a little confusing to me is, how do we determine now what kinds of checks are done in api/ on the client and which are on the server? For example, here you're parsing the URL, but somewhere above (I think it was OIDC or other connector) url.Parse checks were removed. I think we need to be consistent.

Reply:
In my opinion the only checks that should be done in the client should be basic value checks, so I should have moved this to `services` (will now). I want to completely clean up the validation logic across all types in a separate PR to make the separation between client and server validation very clear.